line_profiler/autoprofile/autoprofile.py

```
@@ -92,10 +97,43 @@ def run(script_file, ns, prof_mod, profile_imports=False, as_module=False):
         as_module (bool):
             Whether we're running script_file as a module
     """
-    Profiler = AstTreeModuleProfiler if as_module else AstTreeProfiler
+    @contextlib.contextmanager
+    def restore_dict(d, target=None):
+        copy = d.copy()
+        yield target
+        d.clear()
+        d.update(copy)
+
+    if as_module:
+        Profiler = AstTreeModuleProfiler
+        module_name = modpath_to_modname(script_file)
+        assert module_name is not None
```

Is this something we should check even if python is run with `-O`? It seems like we may want to raise a ModuleNotFound error with a helpful message.

---

Fair point, will do